main-module/src/main/java/ru/practicum/explorewithme/ewmservice/request/service/RequestServiceImpl.java

```diff
@@ -0,0 +1,231 @@
+                        RequestStatus.CONFIRMED) + dto.getRequestIds().size();
+
+                if (countRequestsLimit > event.getParticipantLimit()) {
+                    throw new ConflictException("The participant limit has been reached");
+                }
+                List<Request> requests = updateRequestStatus(dto, eventId);
+
+                EventRequestStatusUpdateResult result = EventRequestStatusUpdateResult.builder()
+                        .confirmedRequests(requests.stream()
+                                .map(participationRequestDtoMapper::toDto)
+                                .collect(Collectors.toList())).build();
+
+                if (countRequestsLimit.equals(event.getParticipantLimit())) {
+                    List<Request> otherRequests = requestDao.findAllByEventIdAndStatus(eventId, RequestStatus.PENDING);
+                    otherRequests.forEach(request -> request.setStatus(RequestStatus.REJECTED));
+                    requestDao.saveAll(otherRequests);
+
+                    result.setRejectedRequests(otherRequests.stream()
+                            .map(participationRequestDtoMapper::toDto)
+                            .collect(Collectors.toList()));
+
+                }
+                return result;
+            } else {
+                List<Request> requests = updateRequestStatus(dto, eventId);
+                return EventRequestStatusUpdateResult.builder()
+                        .confirmedRequests(requests.stream()
+                                .map(participationRequestDtoMapper::toDto)
+                                .collect(Collectors.toList())).build();
+            }
+        } else {
+            List<Request> rejectedRequest = updateRequestStatus(dto, eventId);
+
+            return EventRequestStatusUpdateResult.builder()
+                    .rejectedRequests(rejectedRequest.stream()
+                            .map(participationRequestDtoMapper::toDto)
+                            .collect(Collectors.toList()))
+                    .build();
+        }
+    }
+
+    @Override
+    public Map<Long, Integer> countConfirmedRequests(List<Long> eventIds) {
+        List<Request> eventsByIdInAndStatus = requestDao.findAllByEventIdInAndStatus(eventIds,
+                RequestStatus.CONFIRMED);
+
+        return eventsByIdInAndStatus.stream()
+                .collect(Collectors.groupingBy(
+                        request -> request.getEvent().getId(),
+                        Collectors.collectingAndThen(Collectors.counting(), Long::intValue))
+                );
+    }
+
+    private List<Request> updateRequestStatus(EventRequestStatusUpdateRequest dto, Long eventId) {
+        List<Request> requests = requestDao.findAllByIdInAndEventId(dto.getRequestIds(),
+                eventId);
+        requests.forEach(request -> {
+            if (!request.getStatus().equals(RequestStatus.PENDING)) {
```

Comment:
```suggestion
            if (request.getStatus() != RequestStatus.PENDING) {
```
enum можно сравнивать так, также это nullsafe подход, если у тебя request.getStatus() может вернуть ничего